joomla/joomla-cms
# Smart Search common words assigned too much weight

Common words (also known as "stop words") in Smart Search were being assigned too much weight in search queries and were not flagged as being common.  The reason turned out to be that the default language "*" was not being recognised as matching the language code used in the common words table ("en").  Thus common words were simply not being recognised as such.

Note that this only affects English because no other language has common words at the present time (unless you added them to the database yourself).
### Summary of Changes

This PR amends the FinderIndexerHelper::isCommon method so that "*" is recognised as shorthand for the default language.
### Testing Instructions

It's difficult to construct a search query where it makes a significant difference to the outcome.  In fact, I've given up trying!  The easiest way is to simply look in the #__finder_terms table and notice that the word "the" has dropped from a weight of 0.2 to 0.025 and it has a 1 in the "common" column.  Prior to applying this PR there wouldn't be any terms flagged as common.

Note that you will need to purge and re-index after applying this PR.  Re-indexing without purging will not force the weights to be recalculated.

Fixing this bug is really about correctly labelling common words so as to pave the way for more sophisticated ranking algorithms in the future.
### Documentation Changes Required

None.  This is a bug fix.
